apps/expo/src/context/global-context.tsx

```diff
@@ -49,14 +49,15 @@ const GlobalContextProvider = ({ children }: IGlobalContextProviderProps) => {
         name: clerkUserData.fullName ? clerkUserData.fullName : 'User',
       })
 
-      setUserData({
-        id: response.id,
-        clerkId: response.clerkId,
-        username: response.username,
-        name: response.name!,
-      })
+        setUserData({
+          id: response.id,
+          clerkId: response.clerkId,
+          username: response.username,
+          name: response.name || '',
+        })
+      }
     }
-  }, [clerkUserData])
+  }, [clerkUserData, createUser])
```

Comment:
Update: When removing the brace and comma, run into this error. Not too sure why. Maybe @thienguen has an idea? 
![image](https://github.com/UNLV-CS472-672/2024-S-GROUP3-Barbell/assets/61567471/4b3400be-d3c7-4db6-85ff-7e5c8075472f)

It could be that the clerkUserData is updating itself which causes the function to infinitely run and render resulting in this error?

Reply:
Go grab the correct code from main branch or in this repo, and it should be good.

Reply:
Sounds good, I will check this out again later tonight!